### PR TITLE
feat: support roman numerals

### PR DESCRIPTION
**Pull request by hippietrail**
# Issues 

Not filed as an issue but Roman numerals came up in discussion in #1190

# Description

Adds support for Roman numerals. Not as a `TokenKind` since words, especially `I` can be simultaneously an English word and a Roman numeral, and only analysing the context can reveal which. Instead they are added as part of the orthography flags in the `WordMetadata` that also tracks whether dictionary entries are capitalized, hyphenated, apostrophized, use camel case, etc.

The code first checks that only the letters MDCLXVI are used and they can't be mixed case. If that test passes then a more complex check is done to make sure that the letters are in a sensible order. We could probably go with just the second test but my hunch was that it would be slower or have a higher potential for edge cases. I can remove the preliminary test if that way is preferred.

In the POS tags snapshots they are marked with `#r` in line with `#` for regular numbers and `#d` for decades.

# Demo
image

# How Has This Been Tested?

Unit tests have been added to enforce all the above constraints.

# Checklist

- [x] I have performed a self-review of my own code
- [x] I have added tests to cover my changes

### CODE REVIEW

**Comment by hippietrail** [comment]
> I like this. Good stuff.
> 
> In a separate PR, it might be a good idea to skip spellchecking on Roman numbers whose length is greater than or equal to two.

Or skipped if the roman numerals property is set but none of the POS properties perhaps.

**Comment by elijah-potter** [comment]
Ooh yes, I like that too.

File: harper-core/src/spell/rune/attribute_list.rs [hunk 1]
```diff
@@ -333,9 +333,101 @@ fn check_orthography(word: &MarkedWord) -> OrthFlags {
...
+
+    if !check_roman_group(&mut chars, 'i', 'v', 'x') {
+        return false;
+    }
+
+    if chars.next().is_some() {
+        return false;
+    }
+
+    println!("Valid roman numeral: {}", s);
```
Comment (elijah-potter) [inline harper-core/src/spell/rune/attribute_list.rs]: On second thought, we don't want these functions to print anything.

Comment (hippietrail) [reply harper-core/src/spell/rune/attribute_list.rs]: > On second thought, we don't want these functions to print anything.

Woah good catch! I'm on it...
